### ISSUE TITLE
Contains fails when child content of a node contains string literal or numeric content.

### ISSUE THREAD

**Issue by chriscartlidge** [airbnb/enzyme#102]
When doing a contains on an element such as:

``` javascript
const wrapper = shallow(<ReviewRating reviewTotal='10' reviewRating='2.5' />);
...
wrapper.contains(<span className='star-rating-text'>(10)</span>)
```

The comparison fails due to the "(10)" being represented as an array of chars in the wrappers tree e.g. 

``` javascript
Expected:
{ className: 'star-rating-text', children: '(5)' }

Actual:
{ className: 'star-rating-text', children: [ '(', '5', ')' ] }
```

However, if I use a numeric value under the value 10 I get a false positive...

I have supplied a gist for an easy way to reproduce the issue: (https://gist.github.com/chriscartlidge/f96bf3318524020eff6f)

**Comment by lelandrichardson** [comment]
hmm. This is interesting. There's a lot of different possibilities here...

I do think that in an ideal world

```
<div>foo bar</div>
```
would match
```
<div>{"foo"} {"bar"}</div>
```

But that is going to be kind of difficult to implement. In the eyes of react, this is sort of like comparing:

```
<div><span>foo bar</span></div>
```
with
```
<div><span>foo</span><span> </span><span>bar</span></div>
```
For the sake of our comparison algorithm, we could perhaps write in some exceptions for strings, though.

What is the ideal behavior when comparing `<div>5</div>` with `<div>{5}</div>` though?

**Comment by ljharb** [comment]
In that case, I'd expect children to be a string, and a number, respectively, and just like PropTypes, I would **not** expect them to both be matched irregardless of type. In other words, you can't match a number 5 with a string 5 and vice versa.

**Comment by chriscartlidge** [comment]
I agree with @ljharb that it should treat them as string vs number. I have found a couple of other scenarios where this fails as well.

I'll try and get a few gist up for you.

**Comment by nfcampos** [comment]
fixed by #512, released on 2.8.0
